### PR TITLE
Support pretty printing

### LINKED ISSUE #53: Support pretty printing as an option when writing Ion text

### PR DESCRIPTION
**Pull request by joshterrell805**
Fixes #53

### CODE REVIEW

File: amazon/ion/writer_text.py [hunk 1]
```diff
@@ -341,7 +343,7 @@ def serialize(ion_event):
 _CONTAINER_DELIMITER_MAP = {
     IonType.STRUCT: b',',
     IonType.LIST: b',',
-    IonType.SEXP: b' ',
+    IonType.SEXP: {'normal': b' ', 'pretty': b''},
```
Comment (tgregg) [inline amazon/ion/writer_text.py]: Maybe it's personal preference, but I'd rather keep the delimiter maps as homogenous `IonType` -> `bytes` `dict`s. 

```python
_CONTAINER_DELIMITER_MAP_NORMAL = {
    IonType.STRUCT: b',',
    IonType.LIST: b',',
    IonType.SEXP: b' ',
}
_CONTAINER_DELIMITER_MAP_PRETTY = {
    IonType.STRUCT: b',',
    IonType.LIST: b',',
    IonType.SEXP: b'',
}
```

Then you'd make a `_serialize_container_delimiter_normal` and `serialize_container_delimiter_pretty` and leave `serialize` as it was. 

```python
_serialize_container_delimiter_normal = _serialize_container_factory('delimiter_normal', _CONTAINER_DELIMITER_MAP_NORMAL)
_serialize_container_delimiter_pretty = _serialize_container_factory('delimiter_pretty', _CONTAINER_DELIMITER_MAP_PRETTY)
```

Then in `_raw_writer_coroutine` you'd have something like
```python
serialize_container_delimiter = _serialize_container_delimiter_pretty if pretty else _serialize_container_delimiter_normal
delimiter = serialize_container_delimiter(container_event)
```

File: amazon/ion/simpleion.py [hunk 2]
```diff
@@ -46,7 +46,7 @@
 
 
 def dump(obj, fp, imports=None, binary=True, sequence_as_stream=False, skipkeys=False, ensure_ascii=True,
-         check_circular=True, allow_nan=True, cls=None, indent=None, separators=None, encoding='utf-8', default=None,
+         check_circular=True, allow_nan=True, cls=None, pretty=False, indent='  ', separators=None, encoding='utf-8', default=None,
```
[the remark this answers is not in the document]
Comment (joshterrell805) [reply amazon/ion/simpleion.py]: That makes total sense, thanks for letting me know about that! I'll remove this pretty parameter.

File: amazon/ion/writer_text.py [hunk 3]
```diff
@@ -411,7 +419,7 @@ def _raw_writer_coroutine(depth=0, container_event=None, whence=None):
 
 
 # TODO Add options for text formatting.
-def raw_writer():
+def raw_writer(pretty=False, indent='  '):
```
[the remark this answers is not in the document]
Comment (joshterrell805) [reply amazon/ion/writer_text.py]: Agreed, will do

File: amazon/ion/writer_text.py (hunk 1 above)
Comment (joshterrell805) [reply amazon/ion/writer_text.py]: That works with me, changed in next revision.

**Pull request merged**